static/panes/opt-pipeline.interfaces.ts

```diff
@@ -23,7 +23,9 @@
 // POSSIBILITY OF SUCH DAMAGE.
 
 export interface OptPipelineViewState {
-    selectedFunction: string;
+    lang: string;
+    compiler: string;
+    selectedGroup: string;
```

Comment:
We will need to make this `lang:? string` etc and keep the `selectedFunction` as an optional too and/or just make sure when we deserialise the window state we apply appropriate patching. We can help with that: we have a version to help with it but it's been a _long_ time since we've had to bump it...we are out of pracitce! :)

Reply:
I have added compatibility support to handle older state and made `lang` and `compiler` optional. `selectedFunction` is now migrated to `selectedGroup` as well.

The pipeline view will fallback to previous functionality when the compiler info is missing (because of older state).

Reply:
Thank you so much!